CMakeLists.txt

```diff
@@ -3,69 +3,86 @@ if(${CMAKE_VERSION} VERSION_LESS 3.12)
     cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
 endif()
 
-project(search LANGUAGES CXX CUDA)
+project(search LANGUAGES CXX)
+
+# Check if we can compile CUDA on this system.
+include(CheckLanguage)
+check_language(CUDA)
+
+set(CPU_ONLY OFF CACHE BOOL "Build without GPU support?")
+
+if(CMAKE_CUDA_COMPILER AND NOT CPU_ONLY)
+  set(HAVE_CUDA 1)
+  enable_language(CUDA)
+  add_definitions(-DHAVE_CUDA=1)
+endif()
 
 include(CheckIPOSupported)
 check_ipo_supported(RESULT ipo_supported)
 
-#find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
-find_package( PythonInterp )
-find_package( PythonLibs )
+find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
 
-find_library(CFITSIO_LIBRARY 
+find_library(CFITSIO_LIBRARY
     NAMES fitsio cfitsio libcfitsio
     HINTS lib/
 )
 
 add_subdirectory(lib/pybind11)
 
-set(CMAKE_CXX_STANDARD 11) # set(PYBIND11_CPP_STANDARD -std=c++11)
+set(CMAKE_CXX_STANDARD 11)
 
 include_directories(
     include/
 )
 
-add_library(searchcu STATIC
-    src/kbmod/search/image_kernels.cu
-    src/kbmod/search/kernels.cu
-)
-
-set_target_properties(searchcu PROPERTIES 
-    POSITION_INDEPENDENT_CODE ON
-    CUDA_VISIBILITY_PRESET "hidden"
-    PREFIX "${PYTHON_MODULE_PREFIX}"
-    SUFFIX "${PYTHON_MODULE_EXTENSION}"
-)
-
-if(ipo_supported)
-    set_property(TARGET searchcu PROPERTY INTERPROCEDURAL_OPTIMIZATION TRUE)
-endif()
-
 
+# Create the python module via pybind11.
 pybind11_add_module(search MODULE
     src/kbmod/search/bindings.cpp
 )
 
 set_target_properties(search PROPERTIES
     CXX_VISIBILITY_PRESET "hidden"
-    INTERPROCEDURAL_OPTIMIZATION TRUE
     PREFIX "${PYTHON_MODULE_PREFIX}"
     SUFFIX "${PYTHON_MODULE_EXTENSION}"
 )
-
 if(ipo_supported)
     set_property(TARGET search PROPERTY INTERPROCEDURAL_OPTIMIZATION TRUE)
 endif()
 
-
 target_compile_options(search PRIVATE $<$<COMPILE_LANGUAGE:CXX>:
     -O3
     -fvisibility=hidden
     -fopenmp
 >)
 
 target_link_libraries(search PRIVATE
-    searchcu
-    ${CFITSIO_LIBRARY}
-    -lgomp
+  ${CFITSIO_LIBRARY}
+  -lgomp
 )
+
+
+# If we have CUDA, build the kernel libraries and link them in as well.
+if(HAVE_CUDA)
+  message(STATUS "Building CUDA Libraries")
+  add_library(searchcu STATIC
+      src/kbmod/search/image_kernels.cu
+      src/kbmod/search/kernels.cu
+  )
+
+  set_target_properties(searchcu PROPERTIES
+      POSITION_INDEPENDENT_CODE ON
+      CUDA_VISIBILITY_PRESET "hidden"
+      CUDA_SEPARABLE_COMPILATION ON
+      CUDA_RESOLVE_DEVICE_SYMBOLS ON
+      PREFIX "${PYTHON_MODULE_PREFIX}"
+      SUFFIX "${PYTHON_MODULE_EXTENSION}"
+  )
+  if(ipo_supported)
+      set_property(TARGET searchcu PROPERTY INTERPROCEDURAL_OPTIMIZATION TRUE)
+  endif()
+
+  target_link_libraries(search PRIVATE searchcu)
+else()
+  message(STATUS "Skipping CUDA Libraries")
```

Comment:
This never prints for me. From the manual it seems as it should. I think I remember reading that Python and C++ don't [deal with their prints very nicely](https://pybind11.readthedocs.io/en/stable/advanced/pycpp/utilities.html) in the Pybind docs. Perhaps something similar could be the cause of the lost message. We could punt that out to `setup.py` - in which case we can basically be sure it would work.

Reply:
If you build with `pip install .`, it doesn't show the output of either the cmake messages or python (setup.py).  You can get them if you do the individual steps (such as running cmake).